details.js

```diff
@@ -1,204 +1,763 @@
+	model[dep_ut] = new ScilabBoolean(new data('true', 0, 0), new data('false', 1, 0));
+	
+	var gr_i = new ScilabString(new data("xstringb(orig(1),orig(2),\"CLKSPLIT_f\",sz(1),sz(2));", 0, 0));
+	
+	var block = new standard_define(new ScilabDouble(new data(80, 0, 0), new data(80, 1, 0)), model, new ScilabDouble(), gr_i); // 1 -> 80
+	block[graphics][style] = new ScilabString(new data("CLKSPLIT_f", 0, 0));
+	return block;
+}
+
+
+function ANDLOG_f() {
+	var model = scicos_model();
+	model[sim] = new ScilabString(new data('andlog', 0, 0));
+	model[out] = new ScilabDouble(new data(1, 0, 0));
+	model[out2] = new ScilabDouble(new data(1, 0, 0)); // null -> 1
+	model[evtin] = new ScilabDouble(new data(-1, 0, 0), new data(-1, 1, 0)) // 1, 1 -> -1, -1
+	model[blocktype] = new ScilabString(new data('d', 0, 0));
+	model[firing] = new ScilabDouble();
+	model[dep_ut] = new ScilabBoolean(new data('false', 0, 0), new data('false', 1, 0));
+
+	var gr_i = new ScilabString(new data("xstringb(orig(1),orig(2),\"ANDLOG_f\",sz(1),sz(2));", 0, 0));
+	var block = new standard_define(new ScilabDouble(new data(80, 0, 0), new data(80, 1, 0)), model, new ScilabDouble(), gr_i); // 3 -> 80
+	
+	// Style
+	block[graphics][out_implicit] = new ScilabString(new data("E",0,0));
+	//block.graphics.out_style = "ExplicitOutputPort;align=right;verticalAlign=middle;spacing=10.0;rotation=0";
+	block[graphics][style] = new ScilabString(new data("ANDLOG_f", 0, 0));
+	return block;
+}
+
+function BasicBlock()
+{
+	switch(arguments[0])
+	{
+		case "ANDBLK":
+			var attributes = {
+				style: "ANDBLK",
+				simulationFunctionName: "csuper",
+				simulationFunctionType: "DEFAULT",
+				blockType: "h",
+				interfaceFunctionName: "ANDBLK"
+			};
+			break;
+		case "CONST_m":
+			var attributes = {
+				style: "CONST_m",
+				simulationFunctionName: "cstblk4",
+				simulationFunctionType: "C_OR_FORTRAN",
+				blockType: "d",
+				interfaceFunctionName: "CONST_m"
+			};
+			break;
+	}
+	var options = attributes || new Object();
+	
+	this.angle=options.angle||"";
+	this.blockType=options.blockType||"";
+	this.connectable=options.connectable||"";
+	this.dependsOnT=options.dependsOnT||"";
+	this.dependsOnU=options.dependsOnU||"";
+	this.id=options.id||"";
+	this.interfaceFunctionName=options.interfaceFunctionName||"";
+	this.ordering=options.ordering||"";
+	this.parent=options.parent||"";
+	this.simulationFunctionName=options.simulationFunctionName||"";
+	this.simulationFunctionType=options.simulationFunctionType||"";
+	this.style=options.style||"";
+	this.value=options.value||"";
+	this.vertex=options.vertex||"";
+	this.visible=options.visible||"";
+	
+	switch(arguments[0])
+	{
+		case "ANDBLK":
+			this.realParameters=ANDBLK();
+			break;
+		case "CONST_m":
+			this.realParameters=CONST_m("define");
+			break;
+	}
+}
+
+function ANDBLK() {
+	
+	var andlog=ANDLOG_f("define");
+	andlog[graphics][orig]=new ScilabDouble(new data(194,0,0),new data(133,1,0));
```

Comment:
data function should not take indices as parameters - rather it should find out what type of arguments are passed and *then* assign indices internally to the values.